### PR TITLE
test: Cypress - Modal, Radio widget automation 

### PR DESCRIPTION
**Pull request by Vijetha-Kaja**
## Description

- Automated Modal and Radio widget test cases mentioned in
https://github.com/appsmithorg/TestSmith/issues/2515
https://github.com/appsmithorg/TestSmith/issues/2514

## Type of change

- Automation

## How Has This Been Tested?
- Cypress test runs

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+  it("1. Modal widget functionality", () => {
+    entityExplorer.DragDropWidgetNVerify(draggableWidgets.BUTTON);
+    entityExplorer.DragDropWidgetNVerify(draggableWidgets.MODAL, 300, 300);
+    entityExplorer.SelectEntityByName("Button1");
+    propPane.EnterJSContext("onClick", "{{showModal('Modal1');}}");
+    deployMode.DeployApp(locators._widgetInDeployed(draggableWidgets.BUTTON));
+
+    //Verify that the Modal widget opens correctly when configured on a button click.
+    agHelper.GetNClick(locators._widgetInDeployed(draggableWidgets.BUTTON));
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts:27]: Can `agHelper.ClickButton` be used here?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    agHelper.AssertElementVisibility(
+      locators._widgetInDeployed(draggableWidgets.ICONBUTTON),
+    );
+    agHelper.GetNClick(locators._widgetInDeployed(draggableWidgets.ICONBUTTON));
+    agHelper.AssertElementAbsence(locators._modal);
+
+    //Verify that clicking outside the Modal widget closes it as expected when Quick dismiss is enabled
+    agHelper.GetNClick(locators._widgetInDeployed(draggableWidgets.BUTTON));
+    agHelper.AssertElementVisibility(locators._modal);
+    agHelper.GetElement("body").click(350, 150);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts:40]: Can `agHelper.ClickOutside` be used?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+    agHelper.AssertElementAbsence(locators._modal);
+  });
+
+  it("2. Verify that multiple Modal widgets can be opened simultaneously.", () => {
+    deployMode.NavigateBacktoEditor();
+    entityExplorer.SelectEntityByName("Button1");
+    propPane.ToggleJSMode("onClick", false);
+    propPane.CreateModal("Modal2", "onClick");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts:48]: I see `CreateModal's` 1st param is not used. Please remove that from the method parameters.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+    propPane.CreateModal("Modal3", "onClick");
+    deployMode.DeployApp(locators._widgetInDeployed(draggableWidgets.BUTTON));
+
+    agHelper.GetNClick(locators._widgetInDeployed(draggableWidgets.BUTTON));
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts:52]: Use `ClicButton` here too & all similar places.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+    agHelper.AssertElementVisibility(locators._modal, true, 2);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts:53]: Possible to check Modal Length to be 3?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+    agHelper.GetNClick(
+      locators._widgetInDeployed(draggableWidgets.ICONBUTTON),
+      1,
+    );
+    agHelper.AssertElementVisibility(locators._modal);
+    agHelper.GetNClick(locators._widgetInDeployed(draggableWidgets.ICONBUTTON));
+    agHelper.AssertElementAbsence(locators._modal);
+  });
+
+  it("3. Verify that scroll appears when there are multiple widgets in modal", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts:68]: Which line validates for the scroll in this case?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+        shouldCreateNewJSObj: true,
+      },
+    );
+    entityExplorer.DragDropWidgetNVerify(draggableWidgets.RADIO_GROUP);
+    propPane.EnterJSContext("Options", "{{JSObject1.myFun1()}}");
+
+    deployMode.DeployApp(
+      locators._widgetInDeployed(draggableWidgets.RADIO_GROUP),
+    );
+    agHelper.GetNAssertContains(widgetLocators.radioWidgetContainer, "test");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts:39]: Consider using below to verify the selected state of radio options:
```
agHelper.AssertExistingCheckedState(
      locators._checkboxTypeByOption("test"),
    );
```

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts [hunk 8]
```diff
@@ -0,0 +1,341 @@
...
+    );
+    deployMode.DeployApp(
+      locators._widgetInDeployed(draggableWidgets.RADIO_GROUP),
+    );
+    agHelper.GetElement("@postExecute").then((interception: any) => {
+      agHelper.Sleep();
+      const name = interception.response.body.data.body[0].name;
+      agHelper.GetNAssertContains(widgetLocators.radioWidgetContainer, name);
+    });
+    agHelper.AssertElementLength(widgetLocators.radioBtn, 10);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts:63]: Why/How 10 here? Dint get.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts [hunk 9]
```diff
@@ -0,0 +1,341 @@
...
+      {
+        "label": "Green",
+        "value": "GREEN"
+      }]`,
+    );
+    deployMode.DeployApp(
+      locators._widgetInDeployed(draggableWidgets.RADIO_GROUP),
+    );
+    agHelper.GetNAssertContains(widgetLocators.radioWidgetContainer, "Blue");
+    agHelper.GetNAssertContains(widgetLocators.radioWidgetContainer, "Green");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts:83]: Use below to verify checked state:
```
agHelper.AssertExistingCheckedState(
      locators._checkboxTypeByOption("Blue"),
    );
```

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts [hunk 10]
```diff
@@ -0,0 +1,341 @@
...
+    }]`,
+    );
+    propPane.UpdatePropertyFieldValue("Default selected value", "red");
+    deployMode.DeployApp(
+      locators._widgetInDeployed(draggableWidgets.RADIO_GROUP),
+    );
+    agHelper.AssertExistingCheckedState(
+      widgetLocators.radioBtn + " input",
+      "false",
+    );
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts:188]:
```
agHelper.AssertExistingCheckedState(
      locators._checkboxTypeByOption("red"),
    );
```

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts [hunk 11]
```diff
@@ -0,0 +1,341 @@
...
+    agHelper.AssertPopoverTooltip("Tooltip text");
+    agHelper.AssertElementEnabledDisabled(
+      locators._widgetInDeployed(draggableWidgets.RADIO_GROUP),
+      0,
+      false,
+    );
+    agHelper
+      .GetElement(locators._widgetInDeployed(draggableWidgets.RADIO_GROUP))
+      .invoke("height")
+      .should("be.greaterThan", 130);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts:293]: Use: `agHelper.GetHeight` or `agHelper.GetWidgetCSSHeight`

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts [hunk 12]
```diff
@@ -0,0 +1,341 @@
...
+    agHelper.AssertCSS(
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts:294]: Use `agHelper.GetWidth`

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts (hunk 8 above)
Comment (Vijetha-Kaja) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts:63]: it fetches data from mock user's api with 10 records as parameter

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Radio/Radio2_spec.ts:63]: Got it.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts [hunk 13]
```diff
@@ -0,0 +1,370 @@
...
+    );
+    agHelper.AssertElementVisibility(
+      locators._modal +
+        " " +
+        locators._widgetInDeployed(draggableWidgets.LIST_V2),
+    );
+    agHelper
+      .GetElement(locators._modal)
+      .invoke("scrollTop")
+      .should("be.greaterThan", 400);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts:101]: Can     `agHelper.ScrollTo(locators._modal, "top")`; be used?

Comment (Vijetha-Kaja) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts:101]: no no, i am checking scroll top height here

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Modal/Modal_spec.ts:101]: Ah apologies for the oversight, got it.

File: app/client/cypress/locators/WidgetLocators.ts [hunk 14]
```diff
@@ -102,6 +102,11 @@ export const WIDGETSKIT = {
     "//*[contains(@class, 'bp3-icon-chevron-up')]//parent::button",
   inputWidgetStepDown:
     "//*[contains(@class, 'bp3-icon-chevron-down')]//parent::button",
+  radioWidgetContainer: "[data-testid='radiogroup-container']",
+  radioBtn: ".bp3-radio",
+  radioWidgetLabel: ".radiogroup-label",
+  radioWidgetLabelContainer:
+    "[data-testid='radiogroup-container'] .label-container",
```
Comment (Aishwarya-U-R) [inline app/client/cypress/locators/WidgetLocators.ts:109]: @Vijetha-Kaja These locators are not used? Can we remove those not used?

Comment (Vijetha-Kaja) [reply app/client/cypress/locators/WidgetLocators.ts:109]: All of these are used in the code

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**